src/main/java/duke/FileHandler.java

```diff
@@ -0,0 +1,128 @@
+            System.out.println("ERROR!!");
+        }
+    }
+
+    public void handleLoadTask(ArrayList<Task> taskList, String path) {
+        try {
+            File taskFile = new File(path);
+            Scanner reader = new Scanner(taskFile);
+            while (reader.hasNextLine()) {
+                String data = reader.nextLine();
+                if (data.contains("(by:")) {
+                    loadDeadline(taskList, data);
+                } else if (data.contains("(at: ")) {
+                    loadEvent(taskList, data);
+                } else {
+                    loadTodo(taskList,data);
+                }
+            }
+        } catch (Exception fileInvalid) {
+            try {
+                File file = new File(path);
+                Files.createDirectories(Path.of(file.getParent()));
+                Files.createFile(Path.of(path));
+            } catch (Exception e) {
+                System.out.println("Error");
+            }
+        }
+    }
+
+    public void loadDeadline(ArrayList<Task> taskList, String data) {
+        data = data.substring(9);
+        String[] taskText = data.split("\\(by:");
+        StringBuilder sb = new StringBuilder(taskText[1].strip());
+        sb.deleteCharAt(sb.length() - 1);
+        Deadline deadlineTask = new Deadline(taskText[0].strip(), sb.toString());
+        if (data.contains("[X]")) {
+            deadlineTask.setDone();
+        }
+        taskList.add(deadlineTask);
+    }
+
+    public void loadEvent(ArrayList<Task> taskList, String data) {
+        data = data.substring(9);
+        String[] taskText = data.split("\\(at:");
+        StringBuilder sb = new StringBuilder(taskText[1].strip());
+        sb.deleteCharAt(sb.length() - 1);
+        Event eventTask = new Event(taskText[0].strip(), sb.toString());
+        if (data.contains("[X]")) {
+            eventTask.setDone();
+        }
+        taskList.add(eventTask);
+    }
+
+    public void loadTodo(ArrayList<Task> taskList, String data) {
+        data = data.substring(9);
+        Todo todoTask = new Todo(data);
+        if (data.contains("[X]")) {
+            todoTask.setDone();
+        }
+        taskList.add(todoTask);
+    }
+
+    public void handleLoadTags(ArrayList<Task> taskList, String path) {
+        try {
+            File taskFile = new File(path);
+            Scanner reader = new Scanner(taskFile);
+            while (reader.hasNextLine()) {
+                String data = reader.nextLine();
+                int index = Integer.parseInt(data.substring(0,1));
+                Task currentTask = taskList.get(index-1);
+                data = data.substring(3,data.length()-1);
+                if(!data.isEmpty()) {
+                    String[] tags = data.split(",");
+                    for (String tag : tags) {
+                        String tagDescription = tag.strip();
+                        currentTask.addTag(tagDescription);
+                    }
+                }
```

Comment:
A guard clause can reduce nesting, like so:
```suggestion
                if(data.isEmpty()) {
                    continue;
                }
                String[] tags = data.split(",");
                for (String tag : tags) {
                    String tagDescription = tag.strip();
                    currentTask.addTag(tagDescription);
                }
```